collated/main/A0139923X.md

```diff
@@ -0,0 +1,727 @@
+                eventList.removeTask(target);
+            }else if(dataType.equals("deadline")){
+                todoList.addTask(task);
+                deadlineList.removeTask(target);
+            }
+    		updateFilteredTodoListToShowAll();
+    		indicateTodoListChanged();
+    	}
+    	else if(task instanceof Event) {
+    	    if(dataType.equals("todo")){
+                eventList.addTask(task);
+                todoList.removeTask(target);
+            }else if(dataType.equals("event")){
+                eventList.addTask(task);
+                eventList.removeTask(target);
+            }else if(dataType.equals("deadline")){
+                eventList.addTask(task);
+                deadlineList.removeTask(target);
+            }
+    		updateFilteredEventListToShowAll();
+    		indicateEventListChanged();
+    	}
+    	else if(task instanceof Deadline) {
+    	    if(dataType.equals("todo")){
+    	        deadlineList.addTask(task);
+    	        todoList.removeTask(target);
+    	    }else if(dataType.equals("event")){
+    	        deadlineList.addTask(task);
+                eventList.removeTask(target);
+    	    }else if(dataType.equals("deadline")){
+    	        deadlineList.addTask(task);
+                deadlineList.removeTask(target);
+    	    }
+    		updateFilteredDeadlineListToShowAll();
+    		indicateDeadlineListChanged();
+    	}
+    	
+    	undoer.prepareUndoEdit(target, dataType, task);
+    }
+
+```
+###### /java/seedu/todoList/model/ModelManager.java
+``` java
+    public synchronized void undoneTask(ReadOnlyTask target, String dataType) throws TaskNotFoundException {
+        switch(dataType) {
+            case "todo":
+                todoList.undoneTask(target);
+                indicateTodoListChanged();
+            case "event":
+                eventList.undoneTask(target);
+                indicateEventListChanged();
+            case "deadline":
+                deadlineList.undoneTask(target);
+                indicateDeadlineListChanged();
+        }
+    }
+    
+    @Override
+    public synchronized void deleteTask(ReadOnlyTask target, String dataType) throws TaskNotFoundException {
+    	switch(dataType) {
+    		case "todo":
+    			todoList.removeTask(target);
+    			indicateTodoListChanged();
+    			undoer.prepareUndoDelete(target);
+    		case "event":
+    			eventList.removeTask(target);
+    			indicateEventListChanged();
+    			undoer.prepareUndoDelete(target);
+    		case "deadline":
+    			deadlineList.removeTask(target);
+    			indicateDeadlineListChanged();
+    			undoer.prepareUndoDelete(target);
+    	}
+    }
+
+    @Override
+    public synchronized void addTask(Task task) throws IllegalValueException, UniqueTaskList.DuplicatetaskException {
+    	if(task instanceof Todo) {
+    		todoList.addTask(task);
+    		updateFilteredTodoListToShowAll();
+    		indicateTodoListChanged();
+    		undoer.prepareUndoAdd(task, "todo");
+    	}
+    	else if(task instanceof Event) {
```

Comment:
coding standard violation; `else` or `else if` should be placed between the braces